src/pages/Analytics/Program/helper.js

```diff
@@ -97,3 +125,43 @@ export const createDataStoreGroupRows = datastore => {
 
     return result
 }
+
+export const updateRows = (current, rows) => {
+    const updatedRow = Object.assign({}, rows)
+    const programFound = Object.keys(updatedRow).find(
+        program => program === current.program
+    )
+
+    if (programFound) {
+        const groupFound = Object.keys(updatedRow[programFound].groups).find(
+            group => group === current.group.id
+        )
+
+        const updatedGroups = groupFound
+            ? {
+                  ...updatedRow[programFound].groups,
+                  [groupFound]: [
+                      ...updatedRow[programFound].groups[groupFound],
+                      current,
+                  ],
+              }
+            : {
+                  ...updatedRow[programFound].groups,
+                  [current.group.id]: [current],
+              }
+
+        updatedRow[programFound] = {
+            ...updatedRow[programFound],
+            groups: updatedGroups,
+        }
+    } else {
+        updatedRow[current.program] = {
+            programName: current.programName,
+            groups: {
+                [current.group.id]: [current],
+            },
+        }
+    }
+
+    return updatedRow
```

Comment:
I haven't tested it, but the following should be equivalent:

```suggestion
    const programRow = rows[current.program]
    
    if (programRow) {
        const group = programRow.groups[current.group.id]
        const updatedGroups = {
            ...programRow.groups,
            [current.group.id]: group ? [...group, current] : [current]
        }

        return {
            ...rows,
            [current.program]: {
                ...programRow,
                groups: updatedGroups
            }
        }
    }
    
    return {
        ...rows,
        [current.program]: {
            programName: current.programName,
            groups: {
                [current.group.id]: [current],
            }
        }
    }
```

Reply:
Thank you, it's working